go/pkg/sdk/sdk_test.go

```diff
@@ -70,95 +94,216 @@ func TestSdkConfigDetails(t *testing.T) {
+
+			fmt.Println()
+			fmt.Println("------------------------------ Increasing allowance... ------------------------------")
+			printUserBalance(tc.mockAddress, client)
+			_, err = increaseAllowance(tc.mockAddress, spender, stableAddress, mockAmount, true, privateKey, tc.rpcEndpoint, tc.chainId)
+			if err != nil {
+				log.Fatalf("Failed to increase allowance: %v", err)
+			}
+			fmt.Println("******************************** Allowance increased ********************************")
+
+			initialBalanceUsdc, err := getAssetUserBalance(tc.mockAddress, stableAddress, client, tc.chainId, true)
+			if err != nil {
+				log.Fatalf("Failed to get user's USDC balance: %v", err)
+			}
+			fmt.Println("Initial usdc balance: ", initialBalanceUsdc)
+			initialBalanceSDai, err := getAssetUserBalance(tc.mockAddress, tc.sDai, client, tc.chainId, true)
+			if err != nil {
+				log.Fatalf("Failed to get user's sDai balance: %v", err)
+			}
+
+			fmt.Println("Initial sDai balance: ", initialBalanceSDai)
+
+			fmt.Println()
+			fmt.Println("-------------------------- Creating deposit transaction... --------------------------")
+			printUserBalance(tc.mockAddress, client)
+			txJSON, err := novaSdk.CreateDepositTransaction(tc.stable, tc.mockAddress, mockAmount, mockReferral)
+			if err != nil {
+				log.Fatalf("Failed to call 'CreateDepositTransaction' function: %s", err)
+			}
+			fmt.Println("- Transaction created")
+
+			var tx types.Transaction
+			err = json.Unmarshal([]byte(txJSON), &tx)
+			if err != nil {
+				log.Fatalf("Failed to unmarshal transaction: %v", err)
+			}
+
+			signedTx, err := utils.SignTransaction(&tx, big.NewInt(tc.chainId), privateKey)
+			if err != nil {
+				log.Fatalf("TestSdkCreateDepositTx: %v", err)
+			}
+			fmt.Println("-- Transaction signed")
+
+			err = utils.SendTransaction(signedTx, tc.rpcEndpoint)
+			if err != nil {
+				log.Fatalf("TestSdkCreateDepositTx: %v", err)
+			}
+			fmt.Println("---- Transaction sent")
+			fmt.Println("********************************** Deposit success **********************************")
+
+			newBalanceUsdc, _ := getAssetUserBalance(tc.mockAddress, stableAddress, client, tc.chainId, true)
+			expectedBalanceUsdc := new(big.Int).Sub(initialBalanceUsdc, mockAmount)
+
+			newBalanceSDai, err := getAssetUserBalance(tc.mockAddress, tc.sDai, client, tc.chainId, true)
+			if err != nil {
+				log.Fatalf("Failed to get user's sDai balance: %v", err)
+			}
+			receivedSDai := new(big.Int).Sub(newBalanceSDai, initialBalanceSDai)
+			fmt.Println("Received sDai: ", receivedSDai)
+
+			_, _, executedPrice, _ := novaSdk.GetSlippage(tc.stable, mockAmount)
+			mockAmountFloat := new(big.Float).SetInt(mockAmount)
+			executedPriceFloat := new(big.Float).SetFloat64(executedPrice)
+			reciprocalPrice := new(big.Float).Quo(big.NewFloat(1), executedPriceFloat)
+			expectedReceivedSDaiFloat := new(big.Float).Mul(mockAmountFloat, reciprocalPrice)
+
+			scaleFactor := new(big.Float).SetFloat64(math.Pow(10, 12))
```

Comment:
why 12? it should depend on tokens' decimals. let's compute this 12 using them

Reply:
sDai is 18 decimals, usdc is 6 and we need to multiply it by 12 to get 18, I will find a way to make it generic

Reply:
I don't find the decimals of sDai on constants.go, so I will use vaultDecimals instead. Tell me if I should retrieve it using the SavingsDaiCaller